gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/AbstractGhfsStatisticsSource.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.fs.gcs;
+
+import org.apache.hadoop.fs.statistics.DurationTracker;
+import org.apache.hadoop.fs.statistics.DurationTrackerFactory;
+import org.apache.hadoop.fs.statistics.IOStatisticsSource;
+import org.apache.hadoop.fs.statistics.impl.IOStatisticsStore;
+
+/**
+ * Base class for implementing IOStatistics sources in the gcs module.
+ *
+ * <p>A lot of the methods are very terse, because GHFSInstrumentation has verbose methods of
+ * similar names; the short ones always refer to the inner class and not any superclass method.
+ */
+abstract class AbstractGhfsStatisticsSource implements IOStatisticsSource, DurationTrackerFactory {
+  private IOStatisticsStore ioStatistics;
+
+  protected AbstractGhfsStatisticsSource() {}
+
+  @Override
+  public IOStatisticsStore getIOStatistics() {
+    return ioStatistics;
+  }
+
+  /**
+   * Setter. this must be called in the subclass constructor with whatever
+   *
+   * @param statistics statistics to set
+   */
+  protected void setIOStatistics(final IOStatisticsStore statistics) {
```

Comment:
We usually do not use `final` modifiers for method parameters.

Reply:
Updated